pages/task/read/index.js

```diff
@@ -17,10 +18,26 @@ module.exports = settings => {
     next();
   });
 
+  app.use('/', (req, res, next) => {
+    if (req.task.data.data.profileId && req.task.data.data.establishmentId) {
+      return req.api(`/establishment/${req.task.data.data.establishmentId}/profile/${req.task.data.data.profileId}`)
+        .then(({ json: { data } }) => {
+          req.profile = cleanModel(data);
+        })
+        .then(() => next())
+        .catch(next);
+    }
+    next();
+  });
+
   app.use('/', form(Object.assign({
     configure: (req, res, next) => {
       req.schema = schemaGenerator(req.task);
 
+      const key = req.task.data.model;
+      // move the content[key] properties one up
+      Object.assign(res.locals.static.content, res.locals.static.content[key]);
```

Comment:
what's the intention here? The added comment doesn't describe the code - it isn't being 'moved' it is being copied, and it seems like data manipulation for the sake of data manipulation. This should be a concern of the template - I assume the 'key' or 'model' property is exposed in the template so you could just use it in a Snippet:

```
<Snippet>{`${key}.property`}</Snippet>
```

Reply:
Copy the _content[key]_ properties one up as they are currently in **pil{}** and we want them in **content{}**, as they will be looked up at content level in _asl-components(fieldset)_. The idea is when we have a task of another type - not just pil application - we can just drop in a file with the task's content - e.g. _content/ppl.js_ and require it in _content/index.js_ .

Reply:
why don't we... drumroll... change the content?

Reply:
I agree with @joefitter - let's not do this, and deal with the dynamic lookups on the client.

The only place I can see where that's actually needed is for the page title anyway.

Reply:
probably, I admit I haven't actually looked at how this is used, so I'm whinging about something I don't know anything about

Reply:
No, it is also needed for the fields labels and hints in [fieldset](
        asl-components/src/fieldset/index.jsx
      ).I think Joe means switch by type ?

Reply:
No, it is also needed for the fields labels and hints in [fieldset](
       https://github.com/UKHomeOffice/asl-components/blob/master/src/fieldset/index.jsx
      ).I think Joe means switch by type ?

Reply:
👍 

I nevertheless maintain that it's probably better to define a prefix to do the lookup using the full path rather than copying _all_ of the content around. Especially since _right now_ the only scope is "PIL or not-PIL" but there will also be dependencies on the task type and status that eventually need to be considered.